### PR TITLE
fix(web): modularizes Web's Toggle-style keyboard menu, converts it fully to TS 🧩

### PR DESCRIPTION
**Pull request by jahorton**
This is the first in a series of four PRs focused on converting Web's old UI "modules" into ES-module compatible code.  They still aren't quite literal modules, but it does get us closer.  

Note that this means that the most-frequently used user-testing page now has a functional keyboard-selection menu for desktop form-factors.  This, in turn, enables testing of keyboard-swap behavior... which is pretty dang important to be able to test.

The code is also converted into a more properly-TypeScript format, with the bulk of the code being placed inside of a TS class.  (The original implementation greatly resembled the look of a _compiled_ TS class.)  This gets us actual Intellisense and type-checking for Toggle UI code, which is a win... and the ability to `import type KeymanEngine` allows Toggle UI code to also type-check against the actual engine, which is a _major_ win.

Restoration of additional published API methods and objects was necessary to fully restore the module, so a fair bit of that work is also included here.  Notable is the new `utilApiEndpoint.ts` 'module' - it exists solely to maintain the published API for `keyman.util`.  The UI modules all make considerable use of the endpoint.

----

#### The UI module interface

This adds a simplistic interface expected of all UI modules:  

https://github.com/keymanapp/keyman/blob/d964dafb88cdddb9e520a9fcc07918adf38be1c9/web/src/app/browser/src/uiModuleInterface.ts#L1-L21

Those three members are "enough" on the main engine's side to support UI module integration for the basic desktop UI.  (The touch-oriented `LanguageMenu` doesn't adhere to it yet, but it wouldn't be a difficult undertaking.)

Note that `initialize` isn't exactly a new expectation:

https://github.com/keymanapp/keyman/blob/c652f4c3bbc131154779b20858add971a4741355/web/src/engine/main/dom/domManager.ts#L1706-L1711

For `shutdown`:

https://github.com/keymanapp/keyman/blob/c652f4c3bbc131154779b20858add971a4741355/web/src/engine/main/kmwbase.ts#L199

https://github.com/keymanapp/keyman/blob/c652f4c3bbc131154779b20858add971a4741355/web/src/engine/main/kmwbase.ts#L210-L212

Specifying this in the engine allows the UI modules to declare adherence to the interface, to have that checked at compile-time, and for the engine to know these functions exist without needing the bracket-quote syntax.  I think that's a win.

#### Restoration of hotkey functionality

The Toggle UI happens to be the primary use-case for KMW's 'hotkey' functionality - it actually specifies specific hotkeys within its code.  Rather than leave that out, I went ahead and restored that as part of this PR.

#### Backward compatibility

While I haven't tested this explicitly, the newly-modularized version of the Toggle UI should be backward-compatible with pre-modularized KMW.  The relevant methods that are expected of the UI modules have the same name, and the `keyman.util` root for many utility functions has been restored.  There are no references to any new, module-only fields or methods; only long-established API and a few internally-known anchor points (like `keyman.ui` and `keyman.ui.name`).

It was admittedly quite tempting to replace certain util-method uses with modular imports.  I strongly prefer some of the nicely-encapsulated, managed classes for stylesheet and dom-event maintenance that only exist in the modular version, but using those would have prevented backward compatibility.  It turned out not to be too difficult a goal to aim for, so those concerns won out.

Those "nicely-encapsulated, managed classes" are still used of course, but only via a common instance managed by the `util` object.  It's not _ideal_ from a resource-maintenance standpoint - cleanup only happens on full-engine shutdown, rather than on the level of the UI module itself - but it gets the job done reasonably well... and results in a more-direct correlation with the old form of the code.

----

@keymanapp-test-bot skip

### CODE REVIEW

File: web/src/app/ui/kmwuitoggle.ts [hunk 1]
```diff
@@ -23,682 +34,756 @@ if(!window['keyman']['ui']['name']) {
...
+
+      selectedMenuItem: HTMLAnchorElement = null;
+      updateList = true;
+
+      /**
+       * The numeric handle for an active window.setTimeout call.
+       */
+      updateTimer: number = null;
+
+      keyboardMenu: HTMLUListElement;
```
Comment (jahorton) [inline web/src/app/ui/kmwuitoggle.ts:74]: This one was utilized inline, but was never declared with the other class fields in the original version.  Now that we can do proper type-checking here, the compiler wanted a proper declaration.

File: web/src/app/ui/build.sh [hunk 2]
```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env bash
+#
+
+# set -x
+set -eu
+
```
Comment (mcdurdin) [inline web/src/app/ui/build.sh:6]:
```suggestion
```

File: web/src/app/ui/kmwuitoggle.ts [hunk 3]
```diff
@@ -23,682 +34,756 @@ if(!window['keyman']['ui']['name']) {
    **/
 
   try {
-
     // Declare KeymanWeb, OnScreen Keyboard and Util objects
-    var keymanweb=window['keyman'],util=keymanweb['util'];
-    var dbg=keymanweb['debug'];
+    const keymanweb = keyman;
```
Comment (mcdurdin) [inline web/src/app/ui/kmwuitoggle.ts:38]: TODO: one day let's eliminate this 🤣

File: web/src/app/ui/kmwuitoggle.ts [hunk 4]
```diff
@@ -23,682 +34,756 @@ if(!window['keyman']['ui']['name']) {
...
+    //var dbg=keymanweb['debug'];
+    const util = keymanweb.util;
```
Comment (mcdurdin) [inline web/src/app/ui/kmwuitoggle.ts:40]: Similarly, one day, eliminate, let minification do the minification...

**Review by mcdurdin — APPROVED** [review]

File: web/src/app/ui/kmwuitoggle.ts (hunk 3 above)
Comment (jahorton) [reply web/src/app/ui/kmwuitoggle.ts:38]: I made an issue for this at #8866.

**Pull request merged**